index.js

```diff
@@ -24,71 +24,99 @@ module.exports = (opts = {}, ctx) => ({
       .action(async (dir = '.') => {
         dir = join(process.cwd(), dir)
         try {
-          const nCtx = await dev({
+          const devContext = await dev({
             sourceDir: dir,
             clearScreen: false,
-            theme: opts.theme || '@vuepress/default'
+            theme: options.theme || '@vuepress/default'
           })
+
           logger.setOptions({ logLevel: 3 })
           logger.info(`Start to generate current site to PDF ...`)
+
           try {
-            await generatePDF(nCtx, {
-              port: nCtx.devProcess.port,
-              host: nCtx.devProcess.host,
-              sorter: opts.sorter
+            await generatePDF(devContext, {
+              port: devContext.devProcess.port,
+              host: devContext.devProcess.host,
+              base: devContext.base,
+              options
             })
           } catch (error) {
-            console.error(red(error))
+            logger.error(red(error))
           }
-          nCtx.devProcess.server.close()
+
+          devContext.devProcess.server.close()
           process.exit(0)
-        } catch (e) {
-          throw e
+        } catch (error) {
+          throw error
         }
       })
   }
 })
 
-async function generatePDF(ctx, {
+const applyFilter = (filter) => {
+  if (typeof filter === 'function') {
+    return (page) => filter(page.location, page)
+  }
+
+  return (page) => {
+    return !filter || filter.test(page.location)
+  }
+}
+
+const createOutputFilename = (dest, pluginConfig, fallback) => {
+  if (typeof dest === 'function') {
+    return dest(pluginConfig)
+  }
+
+  if (typeof dest === 'string') {
+    return dest
+  }
+
+  return `${pluginConfig.title || String(fallback)}.pdf`
+}
+
+async function generatePDF(context, {
   port,
   host,
-  sorter,
+  base,
+  // options can be either a single or multiple option objects
+  options: multiOptions,
 }) {
-  const { pages, tempPath, siteConfig } = ctx
+  const { pages, tempPath, siteConfig } = context
   const tempDir = join(tempPath, 'pdf')
-  fs.ensureDirSync(tempDir)
 
-  let exportPages = pages.slice(0)
+  multiOptions = Array.isArray(multiOptions) ? multiOptions : [multiOptions]
 
-  if (typeof sorter === 'function') {
-    exportPages = exportPages.sort(sorter)
-  }
+  fs.ensureDirSync(tempDir)
 
-  exportPages = exportPages.map(page => {
+  const browser = await puppeteer.launch()
+  const browserPage = await browser.newPage()
+
+  // Generate all pages on bulk
+  const exportPages = pages.slice(0).map(page => {
     return {
       url: page.path,
       title: page.title,
```

Comment:
this isn't quite correct as it changes `http://host` to `http:/host`.   Maybe try
```
      location: `http://${host}:${port}${base || ''}${page.path.slice(1)}`,
```